pkg/cli/admin/upgrade/status/mco/layered_node_state.go

```diff
@@ -228,39 +234,50 @@ func isNodeMCDState(node *corev1.Node, state string) bool {
 	return dstate == state
 }
 
-func checkNodeReady(node *corev1.Node) error {
+func checkNodeReady(l *LayeredNodeState) error {
+	node := l.node
 	for i := range node.Status.Conditions {
 		cond := &node.Status.Conditions[i]
 		// We consider the node for scheduling only when its:
 		// - NodeReady condition status is ConditionTrue,
 		// - NodeDiskPressure condition status is ConditionFalse,
 		// - NodeNetworkUnavailable condition status is ConditionFalse.
 		if cond.Type == corev1.NodeReady && cond.Status != corev1.ConditionTrue {
+			l.ReasonOfUnavailability = fmt.Sprintf("node is reporting NotReady=%v", cond.Status)
 			return fmt.Errorf("node %s is reporting NotReady=%v", node.Name, cond.Status)
 		}
 		if cond.Type == corev1.NodeDiskPressure && cond.Status != corev1.ConditionFalse {
+			l.ReasonOfUnavailability = fmt.Sprintf("node is reporting OutOfDisk=%v", cond.Status)
 			return fmt.Errorf("node %s is reporting OutOfDisk=%v", node.Name, cond.Status)
 		}
 		if cond.Type == corev1.NodeNetworkUnavailable && cond.Status != corev1.ConditionFalse {
+			l.ReasonOfUnavailability = fmt.Sprintf("node is reporting NetworkUnavailable=%v", cond.Status)
 			return fmt.Errorf("node %s is reporting NetworkUnavailable=%v", node.Name, cond.Status)
 		}
 	}
 	// Ignore nodes that are marked unschedulable
 	if node.Spec.Unschedulable {
+		l.ReasonOfUnavailability = fmt.Sprintf("node is reporting Unschedulable")
 		return fmt.Errorf("node %s is reporting Unschedulable", node.Name)
 	}
 	return nil
 }
 
-func isNodeReady(node *corev1.Node) bool {
-	return checkNodeReady(node) == nil
+func isNodeReady(l *LayeredNodeState) bool {
+	return checkNodeReady(l) == nil
 }
 
 // isNodeMCDFailing checks if the MCD has unsuccessfully applied an update
-func isNodeMCDFailing(node *corev1.Node) bool {
+func isNodeMCDFailing(l *LayeredNodeState) bool {
+	node := l.node
 	if node.Annotations[CurrentMachineConfigAnnotationKey] == node.Annotations[DesiredMachineConfigAnnotationKey] {
+		l.ReasonOfUnavailability = fmt.Sprintf("current machine config %s is the same as the desired one %s", node.Annotations[CurrentMachineConfigAnnotationKey], node.Annotations[DesiredMachineConfigAnnotationKey])
 		return false
 	}
-	return isNodeMCDState(node, MachineConfigDaemonStateDegraded) ||
-		isNodeMCDState(node, MachineConfigDaemonStateUnreconcilable)
+	if isNodeMCDState(node, MachineConfigDaemonStateDegraded) ||
+		isNodeMCDState(node, MachineConfigDaemonStateUnreconcilable) {
+		return true
+	}
+	l.ReasonOfUnavailability = fmt.Sprintf("node's mcd state %s is neither %s nor %s", node.Annotations[MachineConfigDaemonStateAnnotationKey], MachineConfigDaemonStateDegraded, MachineConfigDaemonStateUnreconcilable)
```

Comment:
nit: we bump into the machine-config operator enough to be able to unpack "MCD" to "machine-config daemon".  But for this UI where we're trying to explain state to possibly non-expert admins, I'd prefer to spell it out, and avoid the acronym.